### PR TITLE
Look for a default config file in $HOME/.svglintrc.js

### PR DESCRIPTION
**Pull request by kcaran**
This application was exactly what I was look for. Thank you!

I'd like to have a global default config file that I can keep with my environment files: $HOME/.svglintrc.js. This pull looks in your home directory for a .svglintrc.js file if there are no other candidates (i.e. in the working directory or any of its parents). To help me I added the name of the configuration file used to the debug output, and incorporated it in the configuration file tests.

### CODE REVIEW

**Comment by ericcornelissen** [comment]
Thank you for the Pull Request @kcaran. I think the idea of adding this make sense. Unfortunately I'll be quite busy for the rest of this month so it might take a while before I can review+merge this.

File: src/cli/config.js [hunk 1]
```diff
@@ -94,7 +98,16 @@ async function getConfigurationFile(filename, folder) {
         }
     }
 
-    return await getDefaultConfigurationFileTraversingParents(folder);
+    // Look at parent directories or, finally, the user's home directory
+    filepath = await getDefaultConfigurationFileTraversingParents(folder);
+    if (!filepath) {
+        const homedir_file = path.join(os.homedir(), ".svglintrc.js");
```
Comment (ericcornelissen) [inline src/cli/config.js:104]: For style consistency, can you use camelCase for this variable name.

File: src/cli/config.js [hunk 2]
```diff
@@ -94,7 +98,16 @@ async function getConfigurationFile(filename, folder) {
...
+        if (await fileExists(homedir_file)) {
+            filepath = homedir_file;
+        }
+    }
+
+    return filepath || false;
```
Comment (ericcornelissen) [inline src/cli/config.js:110]: Can you remove the `|| false`, I don't think it adds anything at this point?

```suggestion
return filepath;
```

File: src/cli/config.js [hunk 3]
```diff
@@ -94,7 +98,16 @@ async function getConfigurationFile(filename, folder) {
         }
     }
 
-    return await getDefaultConfigurationFileTraversingParents(folder);
+    // Look at parent directories or, finally, the user's home directory
```
Comment (ericcornelissen) [inline src/cli/config.js:101]: Could you remove this comment, I think the code is rather explicit about what is happening already.

As an additional suggestion, to make to code (without comment) even more explicit: you could extract the homedir config file-related logic into a separate function and call+return that conditional on whether or not `getDefaultConfigurationFileTraversingParents` returns something.

This could look something like

```js
filepath = await getDefaultConfigurationFileTraversingParents(folder);
if (filepath) {
  return filepath;
}

return await getConfigurationInHomedir();
```

File: test/cli.spec.js [hunk 4]
```diff
@@ -67,6 +67,7 @@ describe("CLI", function(){
 });
 
 describe("Configuration files", function() {
+    let config_file, filename, project_path;
```
Comment (ericcornelissen) [inline test/cli.spec.js:70]: Please use camelCase for variable names

File: test/cli.spec.js [hunk 5]
```diff
@@ -84,35 +85,59 @@ describe("Configuration files", function() {
...
-        const { failed } = await execCliWith(
-            [VALID_SVG, "--config", "test/projects/esm/foo/custom-svglint-config.js"]
+        project_path = "test/projects/esm/foo/";
+        filename = "custom-svglint-config.js";
+        const { failed, stdout } = await execCliWith(
+            [VALID_SVG, "--config", path.join( project_path, filename ), "--debug"]
         );
+        config_file = path.join( project_path, filename ).replace( /\\/g, "\\\\" );
         expect(failed).toBeFalsy();
+        expect(stdout.replace( /\r?\n/g, "" )).toMatch( new RegExp( `Using configuration file: (.*?)${config_file}`) );
```
Comment (ericcornelissen) [inline test/cli.spec.js:95]: I'm not a particular fan of this kind of test, the fact that the tool logs the location of a file gives no guarentee it actually does what it says. I'd ask you to revert these changes, unless you have a different motivation for the approach.

File: test/svgs/attr.test.svg
Comment (ericcornelissen) [inline test/svgs/attr.test.svg:1]: Is there a motivation for these changes? It seems somewhat arbitrary (albeit more like a conventional SVG). Unless you can show why this is in-scope or that it is indisputably necessary, I'd ask you to revert the changes.

**Review by ericcornelissen — CHANGES_REQUESTED** [review]
Despite the number of comments I think the implementation is pretty much good-to-go.

One further ask: can you update the ["Config" section of the docs](https://github.com/birjj/svglint/tree/834abc486612349a9d4f30e8d7990506ddc0d96c#config) to include this lookup.

**Comment by kcaran** [comment]
Thanks for the review. Now I'm on vacation for the next couple of weeks! But I'll make the updates when I get back.

File: src/cli/config.js (hunk 1 above)
Comment (kcaran) [reply src/cli/config.js]: I renamed variables to use camelCase and updated the README.md file.

File: src/cli/config.js (hunk 3 above)
Comment (kcaran) [reply src/cli/config.js:112]: Removed.

File: test/cli.spec.js (hunk 5 above)
Comment (kcaran) [reply test/cli.spec.js]: Reverted this file.

File: test/svgs/attr.test.svg
Comment (kcaran) [reply test/svgs/attr.test.svg:1]: I found your tool because I wanted to ensure that the SVGs on my website were the correct format. The default test SVG is missing required attributes and the <path> element should be self-closing. I used the updated test SVG to create my configuration file (which I plan on at least posting on my website) but I think the linter should have a more accurate representation of the standards.

https://developer.mozilla.org/en-US/docs/Web/SVG/Element/svg
https://www.aaron-gustafson.com/notebook/please-size-your-inline-svgs/

Comment (ericcornelissen) [reply test/svgs/attr.test.svg:1]: I don't think one should expect test SVGs in this project to be written/formatted correctly, after all we'd want to be sure that invalid SVGs are reported.

The main reason I asked to revert it is because I don't want to spend the time reviewing them since they *appear* irrelevant, not least because it blocks the other changes here. Since I can already tell the changes are incomplete for what you're trying to achieve (e.g. `<circle>` should also be self-closing), I'd instead suggest a separate Pull Request for this topic.

File: src/cli/config.js (hunk 3 above)
Comment (ericcornelissen) [reply src/cli/config.js:112]: It appears the comment is still there.

File: README.md [hunk 6]
```diff
@@ -48,7 +48,7 @@ linting.on("done", () => {
 ## Config
 
 In order to specify what should be linted SVGLint must be given a configuration object.
-If you are using the CLI, this configuration object is read from the file specified by `--config`. This defaults to `.svglintrc.js`, which will be searched for up through the directory tree - this is similar to tools such as ESLint.
+If you are using the CLI, this configuration object is read from the file specified by `--config`. This defaults to `.svglintrc.js`, which will be searched for up through the directory tree - this is similar to tools such as ESLint. If no configuration file is found searching up the directory tree, a configuration file located in the user's home directory is used (i.e. "$HOME/.svglintrc.js" on Unix-like systems).
```
Comment (ericcornelissen) [inline README.md:51]: - I think we should put _"this is similar to tools such as ESLint"_ at the end because the homedir behavior is also similar. This also conveniently allows for dropping the connecting text
- I think _"`"$HOME/.svglintrc.js"`_ on Unix-like systems` is an example. It should also work on Windows.
- I'd suggest using code notation for the path.
- I would personally also prefer `~` over `$HOME`, but feel free to keep `$HOME` if you want.

```suggestion
If you are using the CLI, this configuration object is read from the file specified by `--config`. This defaults to `.svglintrc.js`, which will be searched for up through the directory tree, or in the user's home directory (e.g. `~/.svglintrc.js` on Unix-like systems) - this is similar to tools such as ESLint
```

File: test/svgs/attr.test.svg
Comment (kcaran) [reply test/svgs/attr.test.svg:1]: Sure, I understand. I reverted the file.

File: src/cli/config.js (hunk 3 above)
Comment (kcaran) [reply src/cli/config.js]: My mistake.

File: README.md (hunk 6 above)
Comment (kcaran) [reply README.md]: The README should be updated now.

File: src/cli/config.js [hunk 7]
```diff
@@ -75,6 +79,17 @@ async function getDefaultConfigurationFileTraversingParents(folder) {
     }
 }
 
+async function getConfigurationInHomedir() {
```
Comment (ericcornelissen) [inline src/cli/config.js:82]: Just one last request to add documentation for this function.

```suggestion
/**
 * Get the configuration file to use from the home directory.
 * @returns {Promise<String,Boolean>} The path to the configuration file, or false
 */
async function getConfigurationInHomedir() {
```

**Review by ericcornelissen — CHANGES_REQUESTED** [review]

File: test/svgs/attr.test.svg
Comment (ericcornelissen) [reply test/svgs/attr.test.svg:1]: Thank you for your understanding :slightly_smiling_face:

**Review by ericcornelissen — APPROVED** [review]
